### ISSUE TITLE
[HTML5] Cannot call method in a nested struct with $ accessor and then dot notation

### ISSUE THREAD

**Issue by biyectivo** [YoYoGames/GameMaker-Bugs#7537]
### Description
When using HTML5 in the latest runtime (not sure if it happened eariler), if you have a method in a struct within a struct, say:

```
test = {};
		test[$ "key"] = {};
		test.key[$ "nested_method"] = function() { show_debug_message("Hi world"); }
```

you cannot call the method using FIRST the $ accessor and THEN the dot notation on the same line, i.e.

```
test[$ "key"].nested_method(); // Will NOT work in HTML5
```	

Note that you *can* call the method with any of the other three combinations (first dot then $ accessor, or all dots, or all $ accessors). Also this behavior only happens for nested structs and it is exclusive to HTML5 (I tested in Windows, Android and GX.Games without a problem).

The sample project basically contains the code above as well as the other combinations with a small test/assertion struct I quickly whipped up.

### Steps To Reproduce
Either create a new project, new object and put the code above in the Create event, or run the sample attached project. Run them both in HTML5 to reproduce the error and in other platforms to confirm it's HTML5-speciifc.

### Which version of GameMaker are you reporting this issue for?
IDE v2024.8.0.169 Runtime v2024.8.0.216

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

### Which platform(s) are you seeing the problem on?
HTML5

[73930265-61d7-49a5-837d-412e6538b261](https://api.gamemaker.io/api/github/downloads/73930265-61d7-49a5-837d-412e6538b261)

**Comment by rwkay** [comment]
OK looks like the fix for https://github.com/YoYoGames/GameMaker-HTML5/issues/519 was not good... as it is causing this issue I am going to reopen that one and fix this one in the hotfix

**Comment by rwkay** [comment]
Fixed in 2024.8.1

**Comment by YYDan** [comment]
Works correctly again in IDE v2024.8.1.171 Runtime v2024.8.1.218. Verifying.

![image](https://github.com/user-attachments/assets/bbe7522f-c3e3-4403-8c03-262b29be70b0)
